### PR TITLE
Paul

### PR DESCRIPTION
**Pull request by paulmartini**
This repo seems like the right place to put my code to analyze and monitor observing efficiency.

### CODE REVIEW

**Comment by paulmartini** [comment]
Hi @sbailey - I know you are swamped with other stuff. Perhaps you can find another person to do the review of this PR?

**Review by sbailey — CHANGES_REQUESTED** [review]
Thanks Paul.  I put some minor comments inline, plus a separate email comment.  More general comments:
* our desi standards is for bin/ script to _not_ have the `.py` extension, while things that are imported have the `.py` extension.
* In the context of this being installed with other desi code in the future, let's pick less generic script names than `plot_summary`, `plot_sequence`, `update_summary`, and `plot_night`, which could mean a lot of things and lead to confusion.
* similarly, several of the functions in `desisurveyops.utilities` seem specific to this use-case for survey ops, but not necessarily generically useful for other uses.  e.g. `get_twilights` is a nice generic utility function, while `calc_guidelists` seems pretty specific in what it does and what it outputs.  Consider moving some of them into a separate more focused submodule.
* Minor: Consider using `desisurveyops.utils` (like `desisurvey.utils` and `fiberassign.utils`) or `desisurveyops.util` (like `desispec.util`) rather than introducing a 3rd naming `desisurveyops.utilities`.

File: py/desisurveyops/utilities.py [hunk 1]
```diff
@@ -0,0 +1,1006 @@
...
+    Returns
+    -------
+    outdir : string
+        root output directory 
+    '''
+
+    try:
+        outdir = os.environ['DESINIGHTSTATS']
+    except KeyError:
+        outdir = '/global/cfs/cdirs/desi/users/martini/NightStats/'
```
[the remark this answers is not in the document]
Comment (paulmartini) [reply py/desisurveyops/utilities.py]: Thanks for the general comments, Stephen. I've done the following:
- removed the .py on the bin/ scripts and gave them less generic names (plotnightobs, plotobsefficiency, plottimeobs, updateefficiency)
- moved more specific functions from utilities to obsdata.py 
- renamed utilities.py to utils.py to match the naming convention of other repos

I've also changed this specific case to report an error and raise an exception

File: bin/plot_summary.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+if args.summaryfile is None:
+    summaryfile = os.path.join(outdir, args.summaryfile)
+else:
+    summaryfile = os.path.join(outdir, "nightlystats.fits")
+
+if os.path.isfile(summaryfile):
+    hdu = fits.open(summaryfile)
+else:
+    print("Error: {} not found".format(summaryfile))
+    exit
```
[the remark this answers is not in the document]
Comment (paulmartini) [reply bin/plot_summary.py]: Done

File: py/desisurveyops/utilities.py [hunk 3]
```diff
@@ -0,0 +1,1006 @@
...
+    verbose : bool
+        Print verbose output? 
+
+    Returns
+    -------
+    outdir : string
+        root output directory 
+    '''
+
+    datadir = '/global/cfs/cdirs/desi/spectro/data/'
```
[the remark this answers is not in the document]
Comment (paulmartini) [reply py/desisurveyops/utilities.py]: I've switched to `$DESI_SPECTRO_DATA`, then try `$DESI_ROOT`,  and then raise an exception if this fails. I've also renamed the function `get_rawdatadir`.

File: py/desisurveyops/utilities.py [hunk 4]
```diff
@@ -0,0 +1,1006 @@
...
+import matplotlib.pyplot as plt
+from astropy.time import Time
+import json
+import ephem
+import psycopg2
+import pandas as pd
+
+def set_outdir(verbose=False): 
+    '''
+    Set the root directory for the output and a NightlyData/ subdirectory
```
[the remark this answers is not in the document]
Comment (paulmartini) [reply py/desisurveyops/utilities.py]: I've renamed this `get_outdir` and updated the docstring.

File: py/desisurveyops/utilities.py [hunk 5]
```diff
@@ -0,0 +1,1006 @@
...
+        root output directory 
+    '''
+
+    try:
+        outdir = os.environ['DESINIGHTSTATS']
+    except KeyError:
+        outdir = '/global/cfs/cdirs/desi/users/martini/NightStats/'
+
+    if not os.path.isdir(outdir): 
+        os.mkdir(outdir)
```
[the remark this answers is not in the document]
Comment (paulmartini) [reply py/desisurveyops/utilities.py]: Done

File: py/desisurveyops/utilities.py [hunk 6]
```diff
@@ -0,0 +1,1006 @@
...
+
+    # Set observatory lat,long to calculate twilight
+    desi = ephem.Observer()
+    desi.lon = '-111.59989'
+    desi.lat = '31.96403'
+    desi.elev = 2097.
+    desi.date = t.strftime('%Y/%m/%d 7:00')
+
+    # Calculate astronomical twilight times
+    desi.horizon = '-18'
```
[the remark this answers is not in the document]
Comment (paulmartini) [reply py/desisurveyops/utilities.py]: Yes, I'd been meaning to do that and appreciate the nudge/reminder. Now takes the Sun altitude as an argument.

**Comment by paulmartini** [comment]
Hi @sbailey - Thank you for the thorough review. I think I've addressed all of your comments. Would you please take another look when you have time? Thanks!

**Review by sbailey — APPROVED** [review]
Thanks for the updates.  Merging now.

**Pull request merged**